escapades_frontend/src/routes/index.tsx

```diff
@@ -0,0 +1,58 @@
+import { useRoutes } from "react-router-dom";
+import { useAuth } from "../provider/authProvider";
+import { ProtectedRoute } from "./ProtectedRoute";
+import Login from "../pages/Login/Login";
+import Home from "@/pages/Home/Home";
+import Register from "@/pages/Register/Register";
+import Trip from "@/pages/Trip";
+
+const Routes = () => {
+  const { auth } = useAuth();
+  const token = auth.token;
+
+  // Define public routes accessible to all users
+  const routesForPublic = [
+    {
+      path: "/",
+      element: <Home />,
+    },
+  ];
+
+  // Define routes accessible only to authenticated users
+  const routesForAuthenticatedOnly = [
+    {
+      path: "/",
+      element: <ProtectedRoute />, // Wrap the component in ProtectedRoute
+      children: [
+        {
+          path: "/trip",
+          element: <Trip />,
+        },
+      ],
+    },
+  ];
+
+  // Define routes accessible only to non-authenticated users
+  const routesForNotAuthenticatedOnly = [
+    {
+      path: "/register",
+      element: <Register />,
+    },
+    {
+      path: "/login",
+      element: <Login />,
+    },
+  ];
```

Comment:
For my culture, is it the best practice to draw routes in React Router 6?

Reply:
what do you mean ?

Reply:
I used to draw routes using component syntax:
```
<Routes>
  <Route path="/" element={<Dashboard />}>
    <Route
      path="messages"
      element={<DashboardMessages />}
    />
    <Route path="tasks" element={<DashboardTasks />} />
  </Route>
  <Route path="about" element={<AboutPage />} />
</Routes>
```